src/journal/journald-kmsg.c

```diff
@@ -397,11 +397,17 @@ static int dispatch_dev_kmsg(sd_event_source *es, int fd, uint32_t revents, void
 }
 
 int server_open_dev_kmsg(Server *s) {
+        mode_t mode;
         int r;
 
         assert(s);
 
-        s->dev_kmsg_fd = open("/dev/kmsg", O_RDWR|O_CLOEXEC|O_NONBLOCK|O_NOCTTY);
+        if (s->read_kmsg)
+                mode = O_RDWR|O_CLOEXEC|O_NONBLOCK|O_NOCTTY;
+        else
+                mode = O_WRONLY|O_CLOEXEC|O_NONBLOCK|O_NOCTTY;
```

Comment:
we install an EPOLLIN watch on the fd after opening it, in order to get notifications about incoming messages... That needs to be skipped  if reading from /dev/kmsg is disabled, otherwise will fail entirely.

Please *always* test that your patches work correctly and do what they are supposed to do before submitting them here.